include/alpaka/test/event/EventHostManualTrigger.hpp

```diff
@@ -371,21 +371,35 @@ namespace alpaka::test
         {
             ALPAKA_FN_HOST static auto isSupported(DevCudaRt const& dev) -> bool
             {
+#    if CUDA_VERSION < 11070
                 int result = 0;
-                cuDeviceGetAttribute(
-                    &result,
-#    if CUDA_VERSION >= 12000
-                    CU_DEVICE_ATTRIBUTE_CAN_USE_STREAM_MEM_OPS_V1,
+                cuDeviceGetAttribute(&result, CU_DEVICE_ATTRIBUTE_CAN_USE_STREAM_MEM_OPS, dev.getNativeHandle());
+                return result != 0;
 #    else
-                    CU_DEVICE_ATTRIBUTE_CAN_USE_STREAM_MEM_OPS,
+                return true; // Always enabled as of CUDA 11.7
 #    endif
-                    dev.getNativeHandle());
-                return result != 0;
             }
         };
     } // namespace trait
 } // namespace alpaka::test
 
+namespace alpaka::trait::detail
+{
+    // TODO: Replace with cuStreamWaitValue32 once support for CUDA < 12 is dropped.
+    inline auto streamWaitValue(CUstream stream, CUdeviceptr addr, cuuint32_t value, unsigned int flags) -> CUresult
+    {
+        // NVIDIA introduced a new stream memory ops API with CUDA 11.7 (called v2). The corresponding CUDA
+        // functions were suffixed with `_v2`. With CUDA 12.0 v1 of the API was removed and the `_v2` removed
+        // from the new functions. So CUDA <= 11.6 and CUDA >= 12.0 share the same function signature but
+        // internally do different things.
+#    if(CUDA_VERSION < 11070) || (CUDA_VERSION >= 12000)
+        return cuStreamWaitValue32(stream, addr, value, flags);
+#    else
+        return cuStreamWaitValue32_v2(stream, addr, value, flags);
+#    endif
+    }
+} // namespace alpaka::trait::detail
```

Comment:
Someone does not like intentions :rofl:, this code could be added two lines below starting with

```C++
namespace detail
{
...
}
```